lua/weapons/arccw_base/sh_firing.lua

```diff
@@ -300,6 +300,7 @@ function SWEP:PrimaryAttack()
     end
 
     self:DoShootSound()
+	if self.DoExtraFiringEvents then self:DoExtraFiringEvents() end
```

Comment:
Do this sort of thing using Hooks. Use self:GetBuff_Hook("Hook_ExtraFiringEvents") for example.

Reply:
if i do this using hooks, will i still be able to specify this behavior for certain weapons innately? i dont want to have to put like a ghost attachment on, say, the rocket launchers in order for them to have backblast

Reply:
nope, weapons can use the same hooks attachments use!
![image](https://user-images.githubusercontent.com/40924091/98509649-91cef980-222f-11eb-8251-6f7a22269434.png)